### ISSUE TITLE
Builders should try to exit prepared house building site towards more walkable areas

### ISSUE THREAD

**Issue by Kromster80** [Kromster80/knights_province#402]
Sometimes builders can paint themselves into a tile they can not leave:

![image](https://github.com/Kromster80/knights_province/assets/4618474/5d9b087d-8d95-4fa8-8d9b-e737d102f2a7)

They should try to walk out to the WalkConnect area that matches the house doorstep.

**Comment by archergod** [comment]
My suggestion is they should always walk out from Entry Gate of building .. in Real world too when a construction site is enclosed it only have one gate to use .. like Porter only use that gate to put material. So it doesn't need high logic just treat prepared land as building with game ..

**Comment by Kromster80** [comment]
Problem is that the doorstep tile is not guaranteed to be walkable (e.g. it could be blocked by a builder building a road and waiting for the stone)

**Comment by archergod** [comment]
Fair point, I would say then he can look for alternative so why not at start itself.

**Comment by Kromster80** [comment]
Duplicate of #348
